react/utils/pages.ts

```diff
@@ -154,29 +154,36 @@ export function navigate(
   const {
     page,
     params,
-    query,
-    to,
+    query: inputQuery,
+    to: inputTo = '',
     scrollOptions,
     fallbackToWindowLocation = true,
     rootPath,
     replace,
     fetchPage = true,
   } = options
 
-  if (!page && !to) {
+  if (!page && !inputTo) {
     console.error(
       `Invalid navigation options. You should use 'page' or 'to' parameters`
     )
     return false
   }
+  
+  const [to, extractedQuery] = inputTo.split('?')
```

Comment:
:pick: you could check if the type of `inputTo` is `string` before splitting, and throw an error/log it otherwise.

Reply:
She has set a default to empty string up there. And there's typings for the parameters as well

Reply:
@luciannojunior I meant for the case where a non-string value is sent (rather than no value at all).

Reply:
and typings are currently only strict inside the app, right? there's nothing preventing a non-string value from being sent from another app, if this function is exported, is there?

Reply:
(this is just a minor suggestion btw, I'm for doing these verifications, but it shouldn't really be a problem, you can just ignore this)

Reply:
Yes, it makes sense! I will check that. Thanks for the great review